src/cpp/rtps/participant/RTPSParticipantImpl.cpp

```diff
@@ -622,7 +622,11 @@ bool RTPSParticipantImpl::create_writer(
 
     std::lock_guard<std::recursive_mutex> guard(*mp_mutex);
     m_allWriterList.push_back(SWriter);
-    if (!is_builtin)
+    if (is_builtin)
+    {
+        async_thread().wake_up(SWriter);
+    }
+    else
```

Comment:
What about async user writers? We could be checking 
`param.mode == ASYNCHRONOUS_WRITER`

Reply:
It doesn't matter. The goal here is to create the async thread as soon as possible. As the PDP announcer will be always created when creating the participant, the first created writer will always be a builtin one.